resources/components/App.vue

```diff
@@ -26,12 +26,16 @@
 					<card-stack queue="user" />
 				</tab>
 			</tabs>
-		</div>
+		</template>
 
 		<!-- Login message container -->
-		<div v-else>
-			Sorry, you can't see the tabs.
-		</div>
+		<template v-else-if="!user.isAuthenticated">
+			<p v-html="loginMessage" />
```

Comment:
This message involves a URL with query params, which I couldn't get to work with the JS parser. Instead, I'm parsing it in the special page PHP and saving it as a config variable. To get the message HTML to print here, I had to use `v-html`, which as discussed isn't ideal. Is this an appropriate circumstance to use it? @catrope is there a way to get this message to parse in JS so I don't have to do this workaround? See [here](https://github.com/egardner/mediawiki-extensions-MachineVision/blob/master/i18n/en.json#L17)

Reply:
The `{{fullurl:}}` and `{{#Special:}}` magic words aren't supported in the JS parser. It's also not really feasible for the JS parser to support `{{#Special:}}`, because it would need access to the (large) mapping from canonical special page names to localized names.

If you wanted the message to work client-side, you could write the link as `[$1 log in]` and pass in the URL as a parameter (which you'd generate with `mw.util.getUrl( 'Special:Userlogin', { returnto: 'Special:SuggestedTags' } )`). But in this case I think it makes more sense to export the parsed message, as you've done.

Reply:
Thanks for the explanation!